simexpal/base.py

```diff
@@ -1129,25 +1189,33 @@ def output_file_path(self, ext):
 				get_output_file_name(ext, self.instance.shortname, self.repetition))
 
 	def get_status(self):
-		if os.access(self.output_file_path('status'), os.F_OK):
-			with open(self.output_file_path('status'), "r") as f:
-				status_dict = yaml.load(f, Loader=YmlLoader)
-
-			if status_dict['timeout']:
-				return Status.TIMEOUT
-			elif status_dict['signal']:
-				return Status.KILLED
-			elif status_dict['status'] > 0:
-				return Status.FAILED
-			return Status.FINISHED
-		elif os.access(self.output_file_path('out'), os.F_OK):
-			return Status.STARTED
-		elif os.access(self.aux_file_path('run'), os.F_OK):
-			return Status.SUBMITTED
-		elif os.access(self.aux_file_path('lock'), os.F_OK):
-			return Status.IN_SUBMISSION
-
-		return Status.NOT_SUBMITTED
+
+		def _get_last_mod():
```

Comment:
Likewise, this function can be inlined into the caller. (These are micro optimizations, of course, but get_status() is also called a lot and function calls are not entirely free in Python, compared to other languages.)

Reply:
done